src/controllers/controller.cpp

```diff
@@ -2,14 +2,34 @@
 
 #include <QApplication>
 #include <QJSValue>
+#include <QRegularExpression>
 
 #include "controllers/controllerdebug.h"
 #include "controllers/defs_controllers.h"
 #include "moc_controller.cpp"
 #include "util/screensaver.h"
 
-Controller::Controller()
-        : m_pScriptEngineLegacy(nullptr),
+namespace {
+const QRegularExpression invalidLoggingCategoryChars("[^a-z0-9]");
+
+QString sanitizedDevicePrefix(const QString& deviceName) {
+    return QStringLiteral("controller.") + deviceName.toLower().remove(invalidLoggingCategoryChars);
+}
+} // namespace
+
+Controller::Controller(const QString& deviceName)
+        : m_sDeviceName(deviceName),
+          m_logBaseRawString(sanitizedDevicePrefix(deviceName).toLocal8Bit()),
+          m_logBase(m_logBaseRawString.constData()),
+          m_logInputRawString(QString(
+                  sanitizedDevicePrefix(deviceName) + QStringLiteral(".input"))
+                                      .toLocal8Bit()),
+          m_logInput(m_logInputRawString.constData()),
+          m_logOutputRawString(QString(
+                  sanitizedDevicePrefix(deviceName) + QStringLiteral(".output"))
+                                       .toLocal8Bit()),
+          m_logOutput(m_logOutputRawString.constData()),
```

Comment:
This is ugly but it's the only way I got the QLoggingCategory constructor to accept a `const char*`. Trying to use `qPrintable`/`QString::toLocal8Bit().constData()` does not work because the data the `const char*` points to immediately goes out of scope.

Reply:
`QLoggingCategory` stores the passed `const char *` as a member  :scream: 

https://doc.qt.io/qt-5/qloggingcategory.html#QLoggingCategory-1

> Note: category must be kept valid during the lifetime of this object.

Reply:
Yeah, I have no idea why the QLoggingCategory constructor doesn't just copy a QString...